akita/src/internal/crud.ts

```diff
@@ -69,16 +69,28 @@ export class CRUD {
     };
   }
 
-  _update<T extends EntityState>(state: T, ids: ID[], newStateOrFn: object | ((e: Readonly<any>) => object)): T {
+  _update<T extends EntityState>(state: T, ids: ID[], newStateOrFn: object | ((e: Readonly<any>) => object), idKey: string): T {
     const updatedEntities = {};
 
+    let isUpdatingIdKey = false;
```

Comment:
This is a rare case; I'm wondering if it's better to create a dedicate method for this to keep the `update()` method clean. Something like:

```ts
this.store.updateEntityId(oldId, newId);
```

What do u think?

Reply:
I understand the desire to keep the implementation of the method short and nice. Indeed, updating the id field of the entity is an edge case. However, I encourage you to think of customers of Akita. The smaller the Akita's API, the easier it is for the developers to learn it. I'd rather try and avoid introducing a new method on the EntityStore class to handle this edge case.